reset exhaustion and saturation when resetting player

skbeh opened the pull request:
When resetting a player's food level, his exhaustion and saturation are leaved unchanged, which leads to an unfairness because the two values determine how fast the hunger level depletes. This pull request makes the resetting method handles them properly.

Realizedd commented:
Thanks for the fix! If you could change the base branch to the dev branch, I'll merge the request.

skbeh commented:
Rebased.

skbeh committed:
reset exhaustion and saturation when resetting player

When resetting a player's food level, his exhaustion and saturation are leaved unchanged, which leads to an unfairness because the two values determine how fast the hunger level depletes. The commit makes the resetting method handles them properly.
